lib/rdoc/markdown.kpeg

```diff
@@ -992,9 +992,14 @@ Strike = &{ strike? }
          "~~"
          { strike a.join }
 
-# TODO alt text support
-Image = "!" ( ExplicitLink | ReferenceLink ):a
-        { "rdoc-image:#{a[/\[(.*)\]/, 1]}" }
+Image = "!" ExplicitLink:a
+        {
+          # Extract alt text and URL
+          alt_text = a[/\{(.*?)\}/, 1] || ""
+          url = a[/\[(.*?)\]/, 1] || ""
```

Comment:
`a` is a String that always match to these regexp because `ExplicitLink:a` is created by `"{#{l}}[#{s}]"`.
Perhaps it could be more simple if `ExplicitLink` returns structured data
```diff
  ExplicitLink =  Label:l "(" @Sp Source:s Spnl Title @Sp ")"
-                 { "{#{l}}[#{s}]" } # returns string, need to parse this string again...
+                 { { label: l, source: s } } # can return hash
```

(need to change `Link =  ExplicitLink | ReferenceLink | AutoLink` to return a string if ExplicitLink returns hash)

Reply:
Could we do this in a follow-up PR? I think it's out of scope for this PR.

Reply:
Makes sense :+1: